experimental_official_dockerfiles/master/bashrc

```diff
@@ -0,0 +1,61 @@
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# ==============================================================================
+
+# Global setup for Python versions
+
+export PYENV_ROOT="$HOME/.pyenv"
+export PATH="$PYENV_ROOT/bin:$PATH"
+eval "$(pyenv init -)"
+eval "$(pyenv virtualenv-init -)"
+
+# Set up interactive stuff only if running interactively
+[ -z "$PS1" ] && return
+
+export PS1="\[\e[31m\]tf-docker\[\e[m\] \[\e[33m\]\w\[\e[m\] > "
+export TERM=xterm-256color
+alias grep="grep --color=auto"
+alias ls="ls --color=auto"
+
+echo -e "\e[1;31m"
+cat<<TF
+ _____                        ______ _
+|_   _|                       |  ___| |
+  | | ___ _ __  ___  ___  _ __| |_  | | _____      __
+  | |/ _ \ '_ \/ __|/ _ \| '__|  _| | |/ _ \ \ /\ / /
+  | |  __/ | | \__ \ (_) | |  | |   | | (_) \ V  V /
+  \_/\___|_| |_|___/\___/|_|  \_|   |_|\___/ \_/\_/
+
+TF
+echo -e "\e[0;33m"
+
+if [[ "$EUID" -eq 0 ]]; then
+  cat <<WARN
+WARNING: You are running this container as root, which can cause new files in
+mounted volumes to be created as the root user on your host machine.
+
+To avoid this, run the container by specifying your user's userid:
+
+$ docker run -u \$(id -u):\$(id -g) args...
```

Comment:
I think that we need to create an user in the container to make this piratical. We could have side effects with an `uid` without an user.

Reply:
Any idea on how to do that without running into volume mounting permission trouble?

If you add a non-root user to the docker image, then mounted volumes have to be accessible to that user. That can fail in surprising ways.

For example, my workstation has effectively random numbers for UID and GID, so when I mount my tensorflow repo into the running container, the user inside the container can't access those files -- unless it's root, or I chown the files on the host, or I create the internal user with the same UID and GID as I have on my machine. I don't want users to have to fight permissions to run the TF containers.

I tried adding a `tf` user to this container and ran into that exact problem. I'd rather throw out our attempts at user setup and run with root.

It's been an open issue for almost ten years: https://github.com/moby/moby/issues/2259

Reply:
I know that bug,  I meant that there isn't any perfect solution that fit all the cases but I think that having a `1000:1000` user probably will cover many more standard user configurations. The root user could create file that could not be removed on the hosts so it is also not the best solution for a dummy approach.

We could always prompt the user in the script in cases where the passed `u \$(id -u):\$(id -g)` don't match our `1000:1000` user in the image.